### DISCUSSION TITLE
Design Refresh & React/TypeScript Templates?

### DISCUSSION THREAD

**Discussion by davidwesst** [fullstackmb/fullstackmb.github.io#16] [Ideas]
Although we have managed to get the website source code updated and working with 11ty, there is still a lot of other improvements we can make. One things that has been discussed within the community is to refresh the design (which many years old at this point) and update the templating language to use React with TypeScript.

11ty v3.0, which is newly released natively supports both React and TypeScript, but it is in its very early days. Using 11ty is a requirement, as we want the static site generator (SSG) capabilities to generate the site, as it is today.

If anyone has suggestions, or design ideas, they should get posted in this thread for discussion.

**Comment by paiziak-o** [comment]
Was playing a bit with AI and built this website using React. It’s not connected to Eleventy yet.
If you like the design, I’ll try connecting React and Eleventy.

However, I don’t like building with HTML and jQuery because I can’t cover it with unit tests, and it would be a nightmare to maintain in the future if this site will get more pages.

I also want to rework our sponsor section, as I think it’s important to attract other companies to become our sponsors.

This new UI will also require some time to work on new pages like About Us, Sponsors, etc.

What do you think about it?

![image (3)](https://github.com/user-attachments/assets/0e27049e-9e0b-4f55-8f2d-8f7ed6aff94e)

**Reply by NadiiaTysh** [reply]
The design is decent. Just two small questions: can we stretch the content across the width and how about responsiveness of the page? Some interesting templates we can borrow from https://bootstrapmade.com/

**Reply by paiziak-o** [reply]
Thank you @NadiiaTysh, adding mobile view of the page. in my version i am using tailwind.
if the team approve this design, i think we will do a lot of suggestions. 

![image](https://github.com/user-attachments/assets/162020f2-1b17-4c87-be3d-c2ab8b417f9c)

**Reply by davidwesst** [reply]
Curious-- is there a reason why all the images are broken in the screenshots?

**Reply by paiziak-o** [reply]
it is fake data, i didn't connect real backend there. but true, looks badly, will connect it and retake screens.

**Reply by paiziak-o** [reply]
here is the new picture with data 
![modified_image](https://github.com/user-attachments/assets/3e7afd5f-d638-4c0c-a56c-cb63ef858f6f)

**Reply by davidwesst** [reply]
This looks really good-- only minor notes below. 

@fullstackmb/board Can you review and provide feedback. I think this is a nicer layout and it's community driven-- which is awesome.

#### Notes

- Show/Hide the "Become a Sponsor" section. I like that this is available, but right now we are not publicly looking for sponsors. Still, would like to uncomment this widget when we are ready to do so.
- Remove Winnipeg .NET User Group (because that is us). :)
- Social icons in the header and footer don't match (we should keep the ones that are in the current site using Font Awesome v4). Also needs the Slack, Meetup, and "newsletter" icons.
- In the footer, "Join the Conversation" is not readable with current colours (light green on white)

**Reply by paiziak-o** [reply]
thank you for the feedback, i sure, there will be more changes around this.
So, If the board approve the mockup in general, I will fix your comments

**Reply by lunky** [reply]
Hey, looks good to me. Thanks @paiziak-o for stepping up!

**Reply by davidwesst** [reply]
I'm going to call that approved-- @paiziak-o, if you're up for creating a new PR for the design, please do. :)

**Reply by paiziak-o** [reply]
great, will work on it today/tomorrow.

**Reply by NadiiaTysh** [reply]
> This looks really good-- only minor notes below.
> 
> @fullstackmb/board Can you review and provide feedback. I think this is a nicer layout and it's community driven-- which is awesome.
> 
> #### Notes
> * Show/Hide the "Become a Sponsor" section. I like that this is available, but right now we are not publicly looking for sponsors. Still, would like to uncomment this widget when we are ready to do so.
> * Remove Winnipeg .NET User Group (because that is us). :)
> * Social icons in the header and footer don't match (we should keep the ones that are in the current site using Font Awesome v4). Also needs the Slack, Meetup, and "newsletter" icons.
> * In the footer, "Join the Conversation" is not readable with current colours (light green on white)

As for footer this is a good point. We will have to meet regulation requirements regarding accessibility, that take effect on May 1, 2025 https://accessibilitymb.ca/accessibility/compliance.html . There is a website https://webaim.org/resources/contrastchecker/ where contrast can be validated.

**Reply by paiziak-o** [reply]
Here is the demo: https://fullstackmb-github-io-3t2e.vercel.app/

**Reply by paiziak-o** [reply]
@NadiiaTysh  if you have any comments about accessibility, feel free to open an issue, i hope someone will jump on it and fix.

**Reply by NadiiaTysh** [reply]
It looks compliant to requirements. I've checked buttons and text and the contrast ratio passes the test. One minor thing is website version at the bottom - #A1ACBA is the closest colour value that gives contrast ratio 7:1, not sure if it's worthy to bother about.
![image](https://github.com/user-attachments/assets/65923f7c-32e6-4d66-b8c1-fe71643d26ab)

**Comment by paiziak-o** [comment]
i think we can close this discussion @davidwesst
